modin/engines/base/frame/data.py

```diff
@@ -1632,7 +1632,23 @@ def _copartition(self, axis, other, how, sort, force_repartition=False):
         """
         if isinstance(other, type(self)):
             other = [other]
-        if all(o.axes[axis].equals(self.axes[axis]) for o in other):
+
+        is_partitons_aligned = True
```

Comment:
Can we fix the issue without this flag?

Reply:
This flag provides possibility to include only additional conditions in `if-else` branches below instead of adding additional code branch.

Reply:
My concern is that this flag (may be its name) affects logic of the method. You are setting the flag to `True` and it's assumed from its name the partitions are aligned. However, this flag takes a part in the next conditions where it may confuses user/developer. I mean the flag should be `False`, not `True`, in the these conditions in order to do repartitioning.

Reply:
`is_partitons_aligned` -> `is_partitions_aligned`

Reply:
Flag was changed to produce behavior which you propose.